src/homework8/arraylist/ArrayList.java

```diff
@@ -0,0 +1,119 @@
+package homework8.arraylist;
+
+import java.util.Arrays;
+
+public class ArrayList<T> implements List<T> {
+    private int position = 0;
+    private boolean checkIsEmpty;
+    private T[] array;
+
+    public ArrayList() {
+        checkIsEmpty = true;
+        array = (T[]) new Object[2];
```

Comment:
by default, this constructor constructs an empty list with an initial capacity of ten.
better don't use magic number